pkg/apis/serving/v1beta1/inference_service_defaults.go

```diff
@@ -80,6 +80,9 @@ func (isvc *InferenceService) Default() {
 }
 
 func (isvc *InferenceService) DefaultInferenceService(config *InferenceServicesConfig, deployConfig *DeployConfig) {
+	if isvc.ObjectMeta.Annotations == nil {
+		isvc.ObjectMeta.Annotations = map[string]string{}
```

Comment:
only initialize the annotation map when we add some annotations, can move this into the if block. This way we can still keep the nil semantics that there is no annotations.